### PR TITLE
Add sol-netctl API to get extra service properties V4

### PR DESCRIPTION
**Pull request by guchaojie**
This PR is for to improve getting extra service properties for user
All the properties provided by Connman D-Bus Hierarchy

Signed-off-by: guchaojie chao.jie.gu@intel.com

### CODE REVIEW

**Comment by guchaojie** [comment]
@cabelitos @barbieri Sorry for delaying update PR due to Chinese National Holiday. Please have a review , thanks!

File: data/scripts/libsoletta.sym [hunk 1]
```diff
@@ -885,3 +885,20 @@ global:
...
+        sol_netctl_service_is_roaming;
+        sol_netctl_service_get_nameservers;
+        sol_netctl_service_get_timeservers;
+        sol_netctl_service_get_domains;
+        sol_netctl_service_get_proxy;
+        sol_netctl_service_get_provider;
+        sol_netctl_service_get_ethernet;
+local:
+        *;
+};
```
Comment (ceolin) [inline data/scripts/libsoletta.sym]: You have to extends `LIBSOLETTA_1`

File: src/lib/comms/include/sol-netctl.h [hunk 2]
```diff
@@ -300,15 +394,155 @@ int sol_netctl_service_get_network_address(
...
  * desired
  *
- * @return 0-100 on success, -errno on failure.
+ * @return The strength of the services on success, 0 on failure.
  */
 int32_t sol_netctl_service_get_strength(
     const struct sol_netctl_service *service);
 
 /**
+ * @brief Gets the service favorite
```
Comment (ceolin) [inline src/lib/comms/include/sol-netctl.h]: I think would be better

`Checks whether the service is favorite or not`

File: src/lib/comms/include/sol-netctl.h [hunk 3]
```diff
@@ -300,15 +394,155 @@ int sol_netctl_service_get_network_address(
...
+ * @param service The service struct which the service favorite is
+ * desired
+ *
+ * @return @c true if the service is favorite, otherwise @c false.
+ */
+bool sol_netctl_service_is_favorite(
+    const struct sol_netctl_service *service);
+
+/**
+ * @brief Gets the service immutable
```
Comment (ceolin) [inline src/lib/comms/include/sol-netctl.h]: ditto

`Checks whether the service is immutable or not`

**Comment by ceolin** [comment]
the patch looks good, please rebase with the upstream, so I can merge it.

**Comment by cabelitos** [comment]
+1 as well. Please, fix the problems pointed by @ceolin and we're good to go

File: data/scripts/libsoletta.sym (hunk 1 above)
Comment (guchaojie) [inline data/scripts/libsoletta.sym]: If we extend LIBSOLETTA_1  ,  make check-api  in CI test will fail :(

**Pull request closed without merging**